src/ripple/app/ledger/impl/MetaView.cpp

```diff
@@ -482,23 +533,212 @@ Json::Value MetaView::getJson (int) const
 
     ret[jss::nodes] = nodes;
 
-    ret[jss::metaData] = mSet.getJson (0);
+    // VFALCO The meta only exists during apply() now
+    //ret[jss::metaData] = meta.getJson (0);
 
     return ret;
 }
 
 //------------------------------------------------------------------------------
 
+void
+MetaView::apply (BasicView& to,
+    STTx const& tx, TER ter, beast::Journal j)
+{
+    auto const sTx =
+        std::make_shared<Serializer>();
+    tx.add(*sTx);
+
+    std::shared_ptr<Serializer> sMeta;
+
+    if (closed())
+    {
+        TxMeta meta;
+        // VFALCO Shouldn't TxMeta ctor do this?
+        meta.init (tx.getTransactionID(), seq());
+        if (deliverAmount_)
+            meta.setDeliveredAmount(
+                *deliverAmount_);
+
+        Mods newMod;
+        for (auto& it : items_)
+        {
+            auto type = &sfGeneric;
+            switch (it.second.first)
+            {
+            case taaMODIFY:
+            #ifdef META_DEBUG
+                JLOG(j.trace) << "modify " << it.first;
+            #endif
+                type = &sfModifiedNode;
+                break;
+
+            case taaDELETE:
+            #ifdef META_DEBUG
+                JLOG(j.trace) << "delete " << it.first;
+            #endif
+                type = &sfDeletedNode;
+                break;
+
+            case taaCREATE:
+            #ifdef META_DEBUG
+                JLOG(j.trace) << "insert " << it.first;
+            #endif
+                type = &sfCreatedNode;
+                break;
+
+            default: // ignore these
+                break;
+            }
+            if (type == &sfGeneric)
+                continue;
+            auto const origNode =
+                base_.read(keylet::unchecked(it.first));
+            auto curNode = it.second.second;
+            if ((type == &sfModifiedNode) && (*curNode == *origNode))
+                continue;
+            std::uint16_t nodeType = curNode
+                ? curNode->getFieldU16 (sfLedgerEntryType)
+                : origNode->getFieldU16 (sfLedgerEntryType);
+            meta.setAffectedNode (it.first, *type, nodeType);
+            if (type == &sfDeletedNode)
+            {
+                assert (origNode && curNode);
+                threadOwners (meta, origNode, newMod, j);
+
+                STObject prevs (sfPreviousFields);
+                for (auto const& obj : *origNode)
+                {
+                    // go through the original node for
+                    // modified  fields saved on modification
+                    if (obj.getFName().shouldMeta(
+                            SField::sMD_ChangeOrig) &&
+                                ! curNode->hasMatchingEntry (obj))
+                        prevs.emplace_back (obj);
+                }
+
+                if (!prevs.empty ())
+                    meta.getAffectedNode(it.first).emplace_back(std::move(prevs));
```

Comment:
I'd put `prevs' and `finals` into their own scope starting at the decl and ending at the move.